docs/polygon_draw.js

```diff
@@ -175,22 +206,19 @@ class StoredShapeData {
    * After a write completes, checks if there are pending writes and kicks off
    * a new update if so.
    * @return {!Promise} Promise that completes when all currently known writes
-   *    are done (or null if an unexpected error is encountered)
+   *    are done
    */
   finishWriteAndMaybeWriteAgain() {
-    StoredShapeData.pendingWriteCount--;
     const oldState = this.state;
     this.state = StoredShapeData.State.SAVED;
     switch (oldState) {
       case StoredShapeData.State.WRITING:
-        loadingElementFinished(writeWaiterId);
+        this.noteWriteFinished();
```

Comment:
as discussed offline, maybe do this only when all global writes have been finished

Reply:
Done, and added a lot of complexity for good measure.